packages/components/src/components/menupopover/menupopover.utils.ts

```diff
@@ -0,0 +1,21 @@
+import { TAG_NAME as MENUITEM_TAGNAME } from '../menuitem/menuitem.constants';
+import { TAG_NAME as MENUITEMCHECKBOX_TAGNAME } from '../menuitemcheckbox/menuitemcheckbox.constants';
+import { TAG_NAME as MENUITEMRADIO_TAGNAME } from '../menuitemradio/menuitemradio.constants';
+import { TAG_NAME as MENUPOPOVER_TAGNAME } from './menupopover.constants';
+
+/**
+ * Checks if the given menu item is a valid menu item.
+ * @param menuItem - The menu item to check.
+ * @returns True if the menu item is a valid menu item, false otherwise.
+ */
+const isValidMenuItem = (menuItem: Element | null): boolean =>
+  [MENUITEM_TAGNAME, MENUITEMCHECKBOX_TAGNAME, MENUITEMRADIO_TAGNAME]
+    .includes(menuItem?.tagName?.toLowerCase() as typeof MENUITEM_TAGNAME);
```

Comment:
TS error is a bit misleading here:

> TS2345: Argument of type string | undefined is not assignable to parameter of type

Unfortunate `as typeof MENUITEM_TAGNAME` is resolve the TS error, but hide the real problem.

The actual problem is when `menuItem.tagName.toLowerCase` is undefined then you can not call it so this code will fail.

You need to do the "Optional chaining" (`?.`) on the call operator (`()`) as well:

```suggestion
    .includes(menuItem?.tagName?.toLowerCase?.()
```

This will work because if any part of the expression is `undefined` it just return `undefined` and it Ok for the `includes` which will return with `false` in this case